libRALFit/src/ral_nlls_internal.f90

```diff
@@ -1145,11 +1145,15 @@ RECURSIVE SUBROUTINE calculate_step(J,f,hf,g,X,md,md_gn,n,m,use_second_derivativ
        do while (.not. subproblem_success)
           
           if  (num_methods_tried>0) then
-             ! if we're trying this method for the second time,
-             ! return to user
-             if (subproblem_method == options%nlls_method) Go To 100
-             ! only use fallback if the options say it's ok
-             if (.not. options%allow_fallback_method) Go To 100
+             ! If we're using a method for the second time, or
+             ! we're not allowing a fallback, then bail...
+             if ( (subproblem_method == options%nlls_method) .or. & 
+                  (.not. options%allow_fallback_method)) Go To 100
+
+             if (buildmsg(5,.False.,options)) then
+                Write(rec(1), Fmt=3040)
+                Call printmsg(3,.False.,options,1,rec)
```

Comment:
@tyronerees: Should match with buildmsg level 5.
`Call printmsg(5,.False.,options,1,rec)`